cmd/validate.go

```diff
@@ -34,19 +38,80 @@ func newValidateCmd() *cobra.Command {
 
 	f := cmd.Flags()
 
-	// TODO: add validation to the path
 	f.StringVarP(&vc.manifestPath, "manifest", "m", "", "'manifest' asks for the path to the manifest")
 
 	return cmd
 }
 
+// isDirectory determines if a file represented by path is a directory or not
+func isDirectory(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.IsDir(), nil
+}
+
+// getManifests uses fs.WalkDir to retrieve a list of the manifest files within the given manifest path
+func getManifests(f fs.FS, path string) ([]string, error) {
+	var manifests []string
+
+	err := fs.WalkDir(f, path, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("error walking path %s with error: %w", path, err)
+		}
+
+		if !d.IsDir() {
+			log.Debugf("%s is not a directory, appending to manifests", path)
+			manifests = append(manifests, path)
+		} else {
+			log.Debugf("%s is a directory, skipping...", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not walk directory: %w", err)
+	}
+
+	return manifests, nil
+}
+
+// run is our entry point to ValidateManifests
 func (vc *validateCmd) run() error {
+	if vc.manifestPath == "" {
+		return fmt.Errorf("path to the manifests cannot be empty")
+	}
 	ctx := context.Background()
+	var wd, _ = os.Getwd()
+	var testFS = os.DirFS(path.Join(wd, "../pkg/safeguards/tests"))
+
+	isDir, err := isDirectory(vc.manifestPath)
+	if err != nil {
+		return fmt.Errorf("could not determine if given path is a directory: %w", err)
+	}
+
+	var manifests []string
+	if isDir {
+		manifests, err = getManifests(testFS, vc.manifestPath)
```

Comment:
I'm not sure of the purpose of this? You should only be gathering the manifests that the customer specified. This looks to also try and append all of the test manifests, which we don't want